fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -502,6 +502,47 @@ private static SearchParameter getSearchParameterByUrlIfPresent(Map<String, Para
         return result;
     }
 
+    /**
+     * Perform wildcard processing for inclusion search parameters by getting all valid search parameters for the
+     * specified join resource type. Search parameters must have a type of 'reference'.
+     * <p>
+     * If inclusion keyword is ' _include' and a target resource type is specified, search parameter must contain
+     * a matching type.
+     * <p>
+     * If inclusion keyword is '_revinclude', search parameter must have a target resource type matching the resource
+     * type being searched.
+     * 
+     * @param resourceType
+     *            the resource type being searched for
+     * @param joinResourceType
+     *            the resource type for which inclusion search parameters will be returned
+     * @param searchParameterTargetType
+     *            the target resource type for included resources
+     * @param inclusionKeyword
+     *            the inclusion type, either _include or _revinclude
+     * @return
+     *         the inclusion SearchParameters for type {@code resourceType} or empty map if none exist
+     * @throws Exception
+     */
+    private static Map<String, SearchParameter> getInclusionWildcardSearchParameters(String resourceType, String joinResourceType,
+        String searchParameterTargetType, String inclusionKeyword) throws Exception {
+        Map<String, SearchParameter> inclusionSearchParameters = new HashMap<>();
+
+        for (SearchParameter searchParameter : getApplicableSearchParameters(joinResourceType)) {
+            if (searchParameter.getType().getValue().equals("reference") &&
```

Comment:
Looks like SearchParamType.REFERENCE.equals(...) could be done instead against "reference".  Same comment on line 1485.

Reply:
Fixed.